provider/machinepool/machine_pool_resource.go

```diff
@@ -372,6 +384,59 @@ func (r *MachinePoolResource) Create(ctx context.Context, req resource.CreateReq
 	resp.Diagnostics.Append(diags...)
 }
 
+// This handles the "magic" import of the default machine pool, allowing the
+// user to include it in their config w/o having to specifically `terraform
+// import` it.
+func (r *MachinePoolResource) magicImport(ctx context.Context, state *MachinePoolState, resp *resource.CreateResponse) {
+	machinepoolName := state.Name.ValueString()
+	existingState := &MachinePoolState{
+		ID:      types.StringValue(machinepoolName),
+		Cluster: state.Cluster,
+		Name:    types.StringValue(machinepoolName),
+	}
+	state.ID = types.StringValue(machinepoolName)
+
+	notFound, diags := r.readState(ctx, existingState)
+	if notFound {
+		// We disallow creating a machine pool with the default name. This
+		// case can only happen if the default machine pool was deleted and
+		// the user tries to recreate it.
+		diags.AddError(
+			"Can't create machine pool",
+			fmt.Sprintf(
+				"Can't create machine pool for cluster '%s': "+
+					"the default machine pool '%s' was deleted and a new machine pool with that name may not be created. "+
+					"Please use a different name.",
+				state.Cluster.ValueString(),
+				machinepoolName,
+			),
+		)
+	}
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	diags = r.doUpdate(ctx, existingState, state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// the 1AZ-related settings don't apply to the default machine pool
+	if state.MultiAvailabilityZone.IsUnknown() {
+		state.MultiAvailabilityZone = types.BoolNull()
+	}
+	if state.AvailabilityZone.IsUnknown() {
+		state.AvailabilityZone = types.StringNull()
+	}
+	if state.SubnetID.IsUnknown() {
+		state.SubnetID = types.StringNull()
```

Comment:
I think if this attribute is computed - it cannot be null

Reply:
TF considers the values to be set, even if they are null.
I tried removing these lines, and the tests fail:

```
  ╷
  │ Error: Provider returned invalid result object after apply
  │ 
  │ After the apply operation, the provider still indicated an unknown value
  │ for rhcs_machine_pool.worker.availability_zone. All values must be known
  │ after apply, so this is always a bug in the provider and should be reported
  │ in the provider's own repository. Terraform will still save the other known
  │ object values in the state.
  ╵
  ╷
  │ Error: Provider returned invalid result object after apply
  │ 
  │ After the apply operation, the provider still indicated an unknown value
  │ for rhcs_machine_pool.worker.multi_availability_zone. All values must be
  │ known after apply, so this is always a bug in the provider and should be
  │ reported in the provider's own repository. Terraform will still save the
  │ other known object values in the state.
  ╵
  ╷
  │ Error: Provider returned invalid result object after apply
  │ 
  │ After the apply operation, the provider still indicated an unknown value
  │ for rhcs_machine_pool.worker.subnet_id. All values must be known after
  │ apply, so this is always a bug in the provider and should be reported in
  │ the provider's own repository. Terraform will still save the other known
  │ object values in the state.
```

Whereas, they pass w/ the code as written.

Reply:
(pardon the delete and recreate... I thought I ran into a problem, but I just didn't rebuild after experimenting :man_facepalming:)

TF regards Null as a "set" value satisfying the "computed" requirement. When I remove these lines, the subsystem tests (and tests against a live cluster) fail, but as written, both pass.

Error generated by omitting the lines:
```
╷
│ Error: Provider returned invalid result object after apply
│ 
│ After the apply operation, the provider still indicated an unknown value for rhcs_machine_pool.worker.availability_zone. All values must be known after apply, so this is always a bug in the provider and
│ should be reported in the provider's own repository. Terraform will still save the other known object values in the state.
╵
```